lib/typesafe.c

```diff
@@ -407,3 +408,148 @@ struct sskip_item *typesafe_skiplist_pop(struct sskip_head *head)
 
 	return item;
 }
+
+/* heap */
+
+#if 0
```

Comment:
Thought -> Make this turned on as part of `--enable-dev-build`?

Reply:
I thought about that and decided against that because the slowdowns introduced by the hash & heap consistency checks is **_huge_**.  I wrote the checks primarily to help me verify I implemented the data structures correctly; i.e. they have served their purpose and I only left them in as future aid in case someone touches the code.  Enabling the checks is only useful for 2 things:

* confirming some existing doubt that the hash or heap code is not functioning correctly (hopefully this is a very remote possibility, and either way this would start from some suspicion based on some failure)
* trying to find memory corruption issues where something randomly wrote to the hash/heap arrays (for this, valgrind is a much better tool - even if the consistency checks detected a failure, they can provide no information at all on when & how the corruption happened)

Since both of these are IMHO very much edge cases, and considering that we run with dev-build for a lot of testing, I think the cost far outweights the usefulness.